repo/datastore.go

```diff
@@ -439,8 +439,11 @@ type MessageStore interface {
 	Queryable
 
 	// Save a new message
-	Put(messageID, orderID string, mType pb.Message_MessageType, peerID string, msg Message) error
+	Put(messageID, orderID string, mType pb.Message_MessageType, peerID string, msg Message, err string, receivedAt int64, pubkey []byte) error
```

Comment:
At what point is `Put` getting overloaded for all sorts of purposes? Some cases we provide empty keys, some cases we provide 0 timestamp. Plus I don't see any validation happening inside `Put` to prevent us from setting the DB state improperly. We should always approach our abstractions with allowing the least amount of behavior necessary and protecting the inputs for the cases we do allow with extreme prejudice.

Reply:
The empty fields are the ones which make no sense for inbound or outbound messages. For example, received_at makes no sense for outbound messages and pubkey does not make sense for inbound messages.

Reply:
We can resolve this afterward. Tracking: https://github.com/OpenBazaar/openbazaar-go/issues/1855